### ISSUE TITLE
VK_MVK_moltenvk not reported. SDK Version 1.1.121.1.

### ISSUE THREAD

**Issue by Mumsfilibaba** [KhronosGroup/MoltenVK#761]
So I have a problem. Since I updated to SDK version 1.1.121.1 i have roughly 1 fps in my application. Though that maybe modifying the MVKConfiguration could help. Only problem is that I cannot get the VK_MVK_moltenvk extension to be reported.

I have tried to load the vkGetMoltenVKConfigurationMVK function with this piece of code:
```
MVKConfiguration mvkConfig = {};
size_t configSize = sizeof(MVKConfiguration);
            
PFN_vkGetMoltenVKConfigurationMVK GetMoltenVKConfigurationMVK = (PFN_vkGetMoltenVKConfigurationMVK)vkGetInstanceProcAddr(m_Instance, "vkGetMoltenVKConfigurationMVK");
if (GetMoltenVKConfigurationMVK == nullptr)
{
    LOG_DEBUG_ERROR("Vulkan: Failed to load vkGetMoltenVKConfigurationMVK\n");
}
else
{
    GetMoltenVKConfigurationMVK(m_Instance, &mvkConfig, &configSize);
}
```
But it only returns NULL (Since the VK_MVK_moltenvk extension has to be present right?) So I added VK_LAYER_DEBUG=all in my environment and I got this from the output:
```
INFO: Found ICD manifest file /usr/local/share/vulkan/icd.d/MoltenVK_icd.json, version "1.0.0"
DEBUG: Searching for ICD drivers named ../../../lib/libMoltenVK.dylib
DEBUG: Build ICD instance extension list
DEBUG: Instance Extension: VK_KHR_device_group_creation (/usr/local/share/vulkan/icd.d/../../../lib/libMoltenVK.dylib) version 0.0.1
DEBUG: Instance Extension: VK_KHR_get_physical_device_properties2 (/usr/local/share/vulkan/icd.d/../../../lib/libMoltenVK.dylib) version 0.0.2
DEBUG: Instance Extension: VK_KHR_get_surface_capabilities2 (/usr/local/share/vulkan/icd.d/../../../lib/libMoltenVK.dylib) version 0.0.1
DEBUG: Instance Extension: VK_KHR_surface (/usr/local/share/vulkan/icd.d/../../../lib/libMoltenVK.dylib) version 0.0.25
DEBUG: Instance Extension: VK_EXT_debug_report (/usr/local/share/vulkan/icd.d/../../../lib/libMoltenVK.dylib) version 0.0.9
DEBUG: Instance Extension: VK_EXT_debug_utils (/usr/local/share/vulkan/icd.d/../../../lib/libMoltenVK.dylib) version 0.0.1
DEBUG: Instance Extension: VK_EXT_metal_surface (/usr/local/share/vulkan/icd.d/../../../lib/libMoltenVK.dylib) version 0.0.1
DEBUG: Instance Extension: VK_EXT_swapchain_colorspace (/usr/local/share/vulkan/icd.d/../../../lib/libMoltenVK.dylib) version 0.0.4
DEBUG: Instance Extension: VK_MVK_macos_surface (/usr/local/share/vulkan/icd.d/../../../lib/libMoltenVK.dylib) version 0.0.2
DEBUG: Instance Extension: VK_MVK_moltenvk (/usr/local/share/vulkan/icd.d/../../../lib/libMoltenVK.dylib) version 0.0.22
DEBUG: ReadDataFilesInSearchPaths: Searching the following paths for manifest files: /Users/dahlle/Dev/GitHub/Lambda/Build/bin/Debug-macosx-x86_64/Sandbox/vulkan/implicit_layer.d:/etc/xdg/vulkan/implicit_layer.d:/tmp/macos-sdk-build/Vulkan-Loader/build/install/etc/vulkan/implicit_layer.d:/etc/vulkan/implicit_layer.d:/usr/local/share/vulkan/implicit_layer.d:/usr/share/vulkan/implicit_layer.d:/Users/dahlle/.local/share/vulkan/implicit_layer.d
```

However when printing all the available extensions I only get 8 of them and this is that list:
```
[INFO]: [Vulkan] Available Instance-Extensions:
[INFO]:    Instance-Extension 'VK_KHR_device_group_creation'
[INFO]:    Instance-Extension 'VK_KHR_get_physical_device_properties2'
[INFO]:    Instance-Extension 'VK_KHR_get_surface_capabilities2'
[INFO]:    Instance-Extension 'VK_KHR_surface'
[INFO]:    Instance-Extension 'VK_EXT_debug_report'
[INFO]:    Instance-Extension 'VK_EXT_debug_utils'
[INFO]:    Instance-Extension 'VK_EXT_swapchain_colorspace'
[INFO]:    Instance-Extension 'VK_MVK_macos_surface'
```

As you can see there is no VK_MVK_moltenvk extension. 

This makes me believe that there is some bug in what extensions get reported. As you can se neither the  VK_MVK_moltenvk or the VK_EXT_metal_surface gets reported back to the app. If it is not a bug please direct me into what could be wrong. Have scouted the internet for hours now without any result. Have installed the SDK with the install_vulkan.py script.

Should be noted that if I run the VulkanInfo application the same extensions are reported. This makes me believe that it is something that went wrong when I installed the SDK. Could this be a bug or I'm I missing some environment variable or other setup?

Have tried this on a Macbook Pro 2019 macOS 10.15 and a Mac Mini 2014 macOS 10.14.6 and both give me the same result on the 1.1.121.1 SDK.

**Comment by Mumsfilibaba** [comment]
Okay so tested using the runtime variables and that worked fine. They did not solve my problem tho. Disabling the resize (That seems to be the problem in the linked issue) The frametime is still 1000ms.

Profiling the application, it seems like when processing the clearImage commands  takes up most of the time. Have not looked into this that much yet but a quick profiling gives this result. 

Currently I'm clearing the image with vkCmdClearColorImage and vkCmdClearDepthStencilImage, I do not know if this can have something to do with the performance issues. I looked at DiligentEngine and it seems like they do the same thing when clearing their rendertargets, maybe the issue lies here? 

Now I got the environment-variables to work, but calling vkGetMoltenVKConfigurationMVK results in a crash. So I guess that this has something to do with the instance I'm passing in?

**Comment by billhollings** [comment]
Thanks for the additional debugging info.

>Now I got the environment-variables to work, but calling vkGetMoltenVKConfigurationMVK results in a crash. So I guess that this has something to do with the instance I'm passing in?

Yes. If you retrieve the `VkInstance` through the SDK, then whatever Vulkan layers sit between your app and MoltenVK will substitute opaque wrapper objects. Sending these opaque objects directly back into MoltenVK (because the intervening layers don't support the `VK_MVK_moltenvk` extension) will result in crashes. More info on this can be found in the `MoltenVK_Runtime_UserGuide.md` document.

**Comment by billhollings** [comment]
@Mumsfilibaba 

This should be fixed in PR #769...which is also in the 1.1.126 Vulkan SDK. 

Please retest and close if this fixes your issue.

**Comment by Mumsfilibaba** [comment]
This seems so solve the issue.
